### PR TITLE
docs/dev: describe agent design and actor system

### PR DESCRIPTION
**Pull request by lucab**
This adds a development page describing the internal structure of
the `agent` long-running service.
In particular, this introduces the underlying actor model and
provides an overview of the existing actors.

### CODE REVIEW

**Review by travier — APPROVED** [review]
LGTM!

File: docs/development/agent-actor-system.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+layout: default
+nav_order: 1
+parent: Development
+---
+
+# Actor model and agent subsystems
+
+Zincati `agent` command provides a long-running background service which drives the OS through auto-updates.
```
Comment (ashcrow) [inline docs/development/agent-actor-system.md:9]:
```suggestion
The Zincati `agent` command provides a long-running background service which drives the OS through auto-updates.
```

File: docs/development/agent-actor-system.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+[actix]: https://github.com/actix/actix
+
+A general overview on such actor framework is provided by the [Actix book][actix-book], and further API details are covered by its [documentation](docs-rs-actix).
+There is also a companion web-framework called `actix-web`, which is however not relevant in this context.
+
+[actix-book]: https://actix.rs/book/actix/
+[docs-rs-actix]: https://docs.rs/actix
+
+The agent is split into several actors, which encapsulate state and behavior, as shown in the diagram below.
+They can run in parallel and communicate by exchanging messages.
```
Comment (ashcrow) [inline docs/development/agent-actor-system.md:33]: They can or they do?

File: docs/development/agent-actor-system.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+The "rpm-ostree client" actor is responsible for shelling out to the `rpm-ostree` command, in order to interact with the rpm-ostree daemon.
+As client commands require blocking and may take a long time to complete, this entity is implemented as a synchronous actor.
+
+This actor bridges incoming requests (messages) to rpm-ostree actions (CLI commands):
+ * `QueryLocalDeployments` maps to `rpm-ostree status`.
+ * `FinalizeDeployment` maps to `rpm-ostree finalize-deployment`.
+ * `StageDeployment` maps to `rpm-ostree deploy --lock-finalization`.
+
+Those actions are generally requested by the core "update agent" actor via the relevant message, and (processed) results are sent back to it once the task has completed.
```
Comment (ashcrow) [inline docs/development/agent-actor-system.md:66]: nit: `()` here are not needed, but they also do not hurt.

**Review by ashcrow — APPROVED** [review]
Optional items. :+1:

File: docs/development/agent-actor-system.md (hunk 2 above)
Comment (lucab) [reply docs/development/agent-actor-system.md:33]: In general, they can. Some also run in parallel (which is useful in our context), but that depends on the specific arrangement.

**Review by ashcrow — APPROVED** [review]

**Pull request merged**